src/reference/03-Developers-Guide/00.md

```diff
@@ -2,12 +2,93 @@
 out: Developers-Guide.html
 ---
 
-Developer's Guide
------------------
-
-This is the set of documentation about the Architecture of sbt. This
-covers all the core components of sbt as well as the general notion of
-how they all work together. This documentation is suitable for those who
-wish to have a deeper understanding of sbt's core, but already
-understand the fundamentals of `Setting[_]`, `Task[_]` and constructing
-builds.
+  [pickling]: https://github.com/scala/pickling
+  [HowCanIHelp]: http://www.scala-sbt.org/community.html#how-can-I-help
+  [Sbt-Launcher]: Sbt-Launcher.html
+
+Developer's Guide (Work in progress)
+------------------------------------
+
+This is the set of documentation about the Architecture of sbt.
+The target audience of this document is the sbt plugin authors
+and sbt developers. See also [How can I help?][HowCanIHelp]
+
+### Modularization
+
+The process we aim to take for sbt 1.0 is to disassemble sbt into small modules.
+To be clear sbt 0.13's codebase already does consists of numerous subprojects.
+
+Here's what we envision the modilarization process to look like:
+
+- pull apart subprojects by feature
+- make separate Github repositories
+- distinguish public API and internal implemention
+- document the usages
+- clean up the code since we can break bincompat
+- publish to Maven
+
+Here are some of the modules.
+
+#### [sbt/serialization](https://github.com/sbt/serialization)
+
+sbt serialization is an opinionated wrapper around [Scala Pickling][pickling] focused on sbt's usage.
+
+#### [launcher][Sbt-Launcher]
+
+The sbt launcher provides a generic container that can load and run
+programs resolved using the Ivy dependency manager. Sbt uses this as its
+own deployment mechanism. (Hosted at [sbt/launcher](https://github.com/sbt/launcher))
+
+#### [sbt/website](https://github.com/sbt/website)
+
+This website's source.
+
+#### [sbt/sbt](https://github.com/sbt/sbt)
+
+Most of the sbt codebase still lives here.
```

Comment:
We should list out the libraries in here which we plan to modularize:

* sbt/io
* sbt/log
* sbt/completion
* sbt/incremental-compiler (or sbt/zinc)
* sbt/collections